src/app/page/cabinet/cabinet.component.ts

```diff
@@ -126,6 +126,6 @@ export class CabinetComponent implements OnInit {
 
   onDeckRatingClick = (deck: DeckPublic, event: IStarRatingOnClickEvent) => {
     deck.rating = event.rating;
```

Comment:
You mustn't assing the rating to the deck field till it's not successfully set on the server.